src/lib.rs

```diff
@@ -929,6 +927,20 @@ impl BoxConstraints {
     pub fn is_height_bounded(&self) -> bool {
         self.max.height.is_finite()
     }
+
+    /// Check to see if these constraints are legit.
+    pub fn check(&self, name: &str) {
+        if !(0.0 <= self.min.width
+            && self.min.width <= self.max.width
+            && self.max.width <= std::f64::INFINITY)
+            || !(0.0 <= self.min.height
+                && self.min.height <= self.max.height
+                && self.max.height <= std::f64::INFINITY)
+        {
+            eprintln!("Bad BoxConstraints passed to {}:", name);
```

Comment:
Should this be using the (now merged) logging functionality?

Reply:
Ah yes! Would this be:
```
warn!("Bad BoxConstraints passed to {}:", name);
warn!("{:?}", self);
```
?

Reply:
That looks right to me, yes.